src/platform-apps/channels/messages-menu/index.test.tsx

```diff
@@ -1,96 +1,111 @@
       ...props,
     };
 
-    return shallow(<MessageMenu {...allProps} />);
+    return shallow(<MessageMenu {...defaultProps} />);
   };
 
-  it('adds className', () => {
-    const onDelete = jest.fn();
-    const wrapper = subject({ className: 'message-menu', onDelete, canEdit: true });
+  describe('ClassName', () => {
+    it('adds className', () => {
+      const onDelete = jest.fn();
+      const wrapper = subject({ className: 'message-menu', onDelete, canEdit: true });
 
-    expect(wrapper.hasClass('message-menu')).toBe(true);
+      expect(wrapper.hasClass('message-menu')).toBe(true);
+    });
   });
 
-  it('it should renders delete button when icon menu is clicked', () => {
-    const onDelete = jest.fn();
-    const wrapper = subject({ onDelete, canEdit: true });
+  describe('Delete Button', () => {
+    it('should render when canEdit is true and onDelete is provided', () => {
+      const onDelete = jest.fn();
+      const wrapper = subject({ canEdit: true, onDelete }) as ShallowWrapper;
 
-    wrapper.find('IconButton').simulate('click');
+      const dropdownMenu = wrapper.find('DropdownMenu');
+      const items = dropdownMenu.prop('items') as { id: string; onSelect: () => void }[];
+      const deleteItem = items.find((item) => item.id === 'delete');
 
-    expect(wrapper.find('.delete-item').exists()).toBe(true);
-  });
-
-  it('it should show delete modal when delete is clicked', () => {
-    const onDelete = jest.fn();
-    const wrapper = subject({ onDelete, canEdit: true });
+      expect(deleteItem).toBeDefined();
+    });
 
-    wrapper.find('.delete-item').simulate('click');
+    it('should open delete modal when delete button is clicked', () => {
+      const onDelete = jest.fn();
+      const wrapper = subject({ canEdit: true, onDelete });
 
-    expect(wrapper.find('ModalConfirmation').exists()).toBe(true);
-  });
+      const dropdownMenu = wrapper.find('DropdownMenu');
+      const items = dropdownMenu.prop('items') as { id: string; onSelect: () => void }[];
+      const deleteItem = items.find((item) => item.id === 'delete');
 
-  it('it should call delete message when confirm button is clicked', () => {
-    const onDelete = jest.fn();
-    const wrapper = subject({ onDelete, canEdit: true });
+      deleteItem.onSelect();
 
-    wrapper.find('.delete-item').simulate('click');
-
-    (wrapper.find('ModalConfirmation').prop('onConfirm') as any)();
-
-    expect(onDelete).toHaveBeenCalled();
+      const state = wrapper.state() as { deleteDialogIsOpen: boolean };
+      expect(state.deleteDialogIsOpen).toBe(true);
+    });
   });
 
-  it('it should renders edit button when icon menu is clicked', () => {
-    const onEdit = jest.fn();
-    const wrapper = subject({ onEdit, canEdit: true });
+  describe('Edit Button', () => {
+    it('should render when canEdit is true and isMediaMessage is false', () => {
+      const onEdit = jest.fn();
+      const wrapper = subject({ canEdit: true, onEdit }) as ShallowWrapper;
```

Comment:
I'm surprised you had to force this to a type. It should know that from the return value of the function. :thinking: